pkg/model/awsmodel/network.go

```diff
@@ -307,6 +307,10 @@ func (b *NetworkModelBuilder) Build(c *fi.ModelBuilderContext) error {
 					infoByZone[subnetSpec.Zone].NATSubnets = append(infoByZone[subnetSpec.Zone].NATSubnets, subnetSpec)
 					infoByZone[subnetSpec.Zone].HaveIPv6PublicSubnet = true
 				} else {
+					err := addAdditionalRoutes(subnetSpec.AdditionalRoutes, subnetName, publicRouteTable, b.Lifecycle, c)
+					if err != nil {
+						return err
+					}
```

Comment:
Since we only create one RouteTable for all public and utility subnets, attempting to add routes to the public route table from each public/utility SubnetSpec's AdditionalRoutes may result in collisions. For now I think we should consider adding API validation to ensure AdditionalRoutes is not used on public or utility subnets until we come up with a better solution for them (in a separate PR)


@olemarkus any additional concerns here for ipv6/dualstack?

Reply:
I think we can deal with those later should any appear.